### PR TITLE
[fea] add encrypt // decrypt config

### LINKED ISSUE #156: Encrypt .env for webapp project

### PR DESCRIPTION
**Pull request by marco-souza**
# Descrição

This PR adds an encrypted `.env.gpg` file to share credentials
- add scripts to encrypt/decrypt credentials
- update `yarn` configs

This closes #156

### CODE REVIEW

File: middleware.ts [hunk 1]
```diff
@@ -12,8 +16,5 @@ export function middleware(req: NextRequest) {
     }
   }
 
-  return new Response('Auth required', {
-    status: 401,
-    headers: { 'WWW-Authenticate': 'Basic realm="Secure Area"' },
-  });
+  return NextResponse.json({ message: 'Auth required' }, { status: 401 });
```
Comment (marco-souza) [inline middleware.ts:19]: I hate next and this is why:

- https://nextjs.org/docs/messages/returning-response-body-in-middleware
- https://nextjs.org/docs/messages/nested-middleware

It just broke the contract of its APIs in a minor release 🫠

File: package.json [hunk 2]
```diff
@@ -9,6 +9,8 @@
     "build:static": "next build && next export",
     "start": "next start",
     "deploy": "firebase deploy",
+    "encrypt": "gpg -c .env",
```
Comment (marco-souza) [inline package.json:12]: > [!WARNING]  
> This requires you to have `gnupg` installed in your system

File: packages/repositories/firestore.ts [hunk 3]
```diff
@@ -40,7 +41,10 @@ export function makeFirestoreDAO<T>({
   const dbCollection = collection(db, collectionName);
   return {
     add: async (item: T) => {
-      const docRef = await addDoc(dbCollection, item);
+      const docRef = await addDoc(
+        dbCollection,
+        item as WithFieldValue<DocumentData>,
```
Comment (marco-souza) [inline packages/repositories/firestore.ts:46]: I hate js dependency management system and yarn

**Review by nicolasbrandao — APPROVED** [review]
LGTM

File: middleware.ts [hunk 4]
```diff
@@ -1,8 +1,12 @@
 import { NextRequest, NextResponse } from 'next/server';
 
 export function middleware(req: NextRequest) {
-  const basicAuth = req.headers.get('authorization');
+  if (!req.nextUrl.pathname.startsWith('/api')) {
```
Comment (marco-souza) [inline middleware.ts:4]: Routes that do not start with '/api' should not require validation.

**Pull request merged**